homeassistant/components/roborock/image.py

```diff
@@ -38,50 +35,50 @@ async def async_setup_entry(
         for drawable, default_value in DEFAULT_DRAWABLES.items()
         if config_entry.options.get(DRAWABLES, {}).get(drawable, default_value)
     ]
-    entities = list(
-        chain.from_iterable(
-            await asyncio.gather(
-                *(
-                    create_coordinator_maps(coord, drawables)
-                    for coord in config_entry.runtime_data.v1
-                )
+    parser = RoborockMapDataParser(
+        ColorsPalette(), Sizes(), drawables, ImageConfig(), []
+    )
+    await asyncio.gather(
+        *(refresh_coordinators(hass, coord) for coord in config_entry.runtime_data.v1)
+    )
+    async_add_entities(
+        (
+            RoborockMap(
+                f"{coord.duid_slug}_map_{map_info.name}",
+                coord,
+                map_info.flag,
+                map_info.name,
+                parser,
             )
-        )
+            for coord in config_entry.runtime_data.v1
+            for map_info in coord.maps.values()
+        ),
+        True,
```

Comment:
Do you already have data? Otherwise you can remove the `True`

Reply:
No, the most recent change is to stop fetching images up front, and fetch them lazilty.

After thinking about it more, I think this should probably schedule a state update in `async_added_to_hass` instead though.  The motivation is to speed up startup time without blocking on this slow call, and i realize that just adding `True` here probably doesn't do that.